### ISSUE TITLE
TSLint slow

### ISSUE THREAD

**Issue by it6** [SublimeLinter/SublimeLinter-tslint#34]
After some debugging,

Looks like on every single change `tslint` is checking for errors in entire project using `--project` flag, which is causing the linter to be slow. I work on a big `monorepo` with multiple projects which is even worse.

For example I ran this in terminal 
**5-7 seconds**
```/Users/kotamrs/sk/work/redbox_spa/myadp/node_modules/.bin/tslint /Users/kotamrs/sk/work/redbox_spa/myadp/src/modules/espresso/components/tabs/segmented-tabs.component.ts --config /Users/kotamrs/sk/work/redbox_spa/myadp/tslint.json --project /Users/kotamrs/sk/work/redbox_spa/myadp```

vs

**less than 1 second without the --project flag**
```/Users/kotamrs/sk/work/redbox_spa/myadp/node_modules/.bin/tslint /Users/kotamrs/sk/work/redbox_spa/myadp/src/modules/espresso/components/tabs/segmented-tabs.component.ts --config /Users/kotamrs/sk/work/redbox_spa/myadp/tslint.json /Users/kotamrs/sk/work/redbox_spa/myadp```

**Comment by kaste** [comment]
In that case you can, don't know if you know, set project to false in your settings. 

There is also PR here open which would remove our previous defaults. But I don't use tslint so cannot judge what a good default here would be.

**Comment by kaste** [comment]
You have the SublimeLinter user settings. Somewhere there is a linters setting. In that you 

‘‘‘
linters: {"tslint": {"project": false}}

‘‘‘

(I'm on mobile until Wednesday at least so sorry if that isn't 100% correct.)

**Comment by frankie567** [comment]
Following @kaste recommendation, here is the correct SublimeLinter user settings to make `tslint` works only on the current file (it seemed that `false` wasn't playing nice with `tslint`):

```json
// SublimeLinter Settings - User
{
  "linters": {
    "tslint": {
      "project": ""
    }
  }
}
```

I think this should be the default for the plugin.

Regards!

**Comment by braver** [comment]
Fixed in 1.6
